Fix scaling list parsing when chroma_format_idc is equal to 3

kradhub opened the pull request:
In this case we can have up to 12 scaling list according to specification
section 7.3.2.1.1.

aizvorski commented:
@kradhub Great catch!  

I think this will also need the following change in h264_stream.h to not overrun buffers:

```
    int seq_scaling_matrix_present_flag;
      int seq_scaling_list_present_flag[12];
      int ScalingList4x4[6][16];
      int UseDefaultScalingMatrix4x4Flag[6];
      int ScalingList8x8[6][64];
      int UseDefaultScalingMatrix8x8Flag[6];
```

Does that look right to you?

I'll merge this PR after a few little changes needed in the SVC code (otherwise it will be a merge conflict).

kradhub commented:
> I think this will also need the following change in h264_stream.h to not overrun buffers:
> 
>     int seq_scaling_matrix_present_flag;
>       int seq_scaling_list_present_flag[12];
>       int ScalingList4x4[6][16];
>       int UseDefaultScalingMatrix4x4Flag[6];
>       int ScalingList8x8[6][64];
>       int UseDefaultScalingMatrix8x8Flag[6];
> 
> Does that look right to you?

Indeed. I updated the patch and I rebased it on the current master.
